pkgs/development/libraries/xdg-desktop-portal-shana/default.nix

```diff
@@ -0,0 +1,43 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, meson
+, ninja
+, xdg-desktop-portal
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "xdg-desktop-portal-shana";
+  version = "0.3.9";
+
+  src = fetchFromGitHub {
+    owner = "Decodetalkers";
+    repo = "xdg-desktop-portal-shana";
+    rev = "v${version}";
+    sha256 = "cgiWlZbM0C47CisR/KlSV0xqfeKgM41QaQihjqSy9CU=";
+  };
+
+  nativeBuildInputs = [
+    meson
+    ninja
+  ];
+
+  buildInputs = [
+    xdg-desktop-portal
+  ];
+
+  configurePhase = "";# needed for letting meson run
```

Comment:
I'm intrigued :laughing: So this prevents `./configure` from running at all? But which part of the derivation starts `meson` then?

Reply:
Sorry I dont' really know what I'm doing. I saw in the meson shell hook that it is applied as configurePhase just if the configurePhase variable is empty (lenght zero), but buildRustPackage sets it as
 ```
runHook preConfigure
runHook postConfigure
```
, unless it is overwritten (what I'm doing).

Reply:
```suggestion
  # Needed for letting meson run. rustPackage will overwrite it otherwise.
  configurePhase = "";
```